src/include/buffer.h

```diff
@@ -297,11 +309,11 @@ class CEPH_BUFFER_API buffer {
 
       void advance(int o);
       void seek(unsigned o);
-      char operator*();
-      iterator& operator++();
-      ptr get_current_ptr();
+      char operator*() const;
+      iterator_impl& operator++();
+      ptr get_current_ptr() const;
```

Comment:
@gregsfortytwo, yes, we do not use 
```c
*bl.begin() = 'c';
```
to modify a byte.

if you think we have future needs for this. i will change the const-ness.